### PR TITLE
Some gp wrapper function should not execute on entrydb QE

### PR DESCRIPTION
**Pull request by interma**
There are some gp functions (e.g. `gp_tablespace_location`) in src/backend/catalog/system_views.sql. They are designed to run on master and dispatch internal logic to segment.

But in some situations, they are executed on the entrydb QE, it will cause error result.

For example:
the definition of `gp_tablespace_location`
```
-- pg_tablespace_location wrapper functions to see Greenplum cluster-wide tablespace locations
CREATE FUNCTION gp_tablespace_segment_location (IN tblspc_oid oid, OUT gp_segment_id int, OUT tblspc_loc text)
AS 'SELECT pg_catalog.gp_execution_segment() as gp_segment_id, * FROM pg_catalog.pg_tablespace_location($1)'
LANGUAGE SQL EXECUTE ON ALL SEGMENTS;

CREATE FUNCTION gp_tablespace_location (IN tblspc_oid oid, OUT gp_segment_id int, OUT tblspc_loc text)
RETURNS SETOF RECORD
AS
  'SELECT * FROM pg_catalog.gp_tablespace_segment_location($1)
   UNION ALL
   SELECT pg_catalog.gp_execution_segment() as gp_segment_id, * FROM pg_catalog.pg_tablespace_location($1)'
LANGUAGE SQL EXECUTE ON COORDINATOR;
```
an error result repo
```
create table test (gp_segment_idd int, tblspc_loc text);

# get all tablespace location in DBs
issues=# select * from gp_tablespace_location(1663);
 gp_segment_id | tblspc_loc
---------------+------------
             1 |
             2 |
             0 |
            -1 |
(4 rows)

# but only two tuple here
issues=# insert into test select * from gp_tablespace_location(1663);
INSERT 0 2
issues=# select * from test;
 gp_segment_idd | tblspc_loc
----------------+------------
             -1 |
             -1 |
(2 rows)

```

Checked the plan:
```
issues=# explain insert into test select * from gp_tablespace_location(1663);
                                         QUERY PLAN
--------------------------------------------------------------------------------------------
 Insert on test  (cost=0.25..23.58 rows=333 width=36)
   ->  Redistribute Motion 1:3  (slice1)  (cost=0.25..23.58 rows=333 width=36)
         Hash Key: gp_tablespace_location.gp_segment_id
         ->  Function Scan on gp_tablespace_location  (cost=0.25..10.25 rows=1000 width=36)
 Optimizer: Postgres query optimizer
(5 rows)
```
The reason is the select operation runs in the entryDB QE, so cannot dispatch its internal logic.

So we should report an error message to user, follow the previous similar case: https://github.com/greenplum-db/gpdb/pull/11178.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by interma** [comment]
init cluster failed, reason is 
```
running bootstrap script ... LOG:  gp_role forced to 'utility' in single-user mode
ok
performing post-bootstrap initialization ... LOG:  gp_role forced to 'utility' in single-user mode
2022-02-13 07:21:28.982484 UTC,,,p13669,th-548980608,,,,0,,cmd166,seg-10000,,,,sx1,"FATAL","42704","language ""plpgsql"" does not exist",,"Use CREATE EXTENSION to load the language into the database.",,,,"-- pg_tablespace_location wrapper functions to see Greenplum cluster-wide tablespace locations
CREATE FUNCTION gp_tablespace_segment_location (IN tblspc_oid oid, OUT gp_segment_id int, OUT tblspc_loc text)
RETURNS SETOF RECORD AS
$$
....
```

So cannot use plpgsql in initdb, but we need to judge error in sql, have to find a new way to detect ...

**Comment by interma** [comment]
> So cannot use plpgsql in initdb, but we need to judge error in sql, have to find a new way to detect ...

Move the functions to cdb_schema which located after load_plpgsql(), then it works and all test passed.
https://github.com/greenplum-db/gpdb/blob/70f50245a0de014adfc30c04b33a32dce6da08ed/src/bin/initdb/initdb.c#L3237-L3254

**Comment by interma** [comment]
Please note: my modification is in catalog sql, does it change the catalog interface?

File: src/backend/catalog/cdb_schema.sql [hunk 1]
```diff
@@ -46,4 +46,62 @@ GRANT SELECT ON pg_catalog.gp_distributed_log TO PUBLIC;
 
 ALTER RESOURCE QUEUE pg_default WITH (priority=medium, memory_limit='-1');
 
+
+-- pg_tablespace_location wrapper functions to see Greenplum cluster-wide tablespace locations
+CREATE FUNCTION gp_tablespace_segment_location (IN tblspc_oid oid, OUT gp_segment_id int, OUT tblspc_loc text)
```
Comment (Aegeaner) [inline src/backend/catalog/cdb_schema.sql:51]: shall we use `CREATE OR REPLACE FUNCTION` to keep consistent with previous definition on views above?

**Review by kainwen — APPROVED** [review]
For master  I think this fix is OK.

For 6X maybe we can use some banned-list when exec a function? Maybe a discussion on gpdb-dev will be good to hear more ideas.

**Comment by interma** [comment]
Thanks Zhenghua, merged this PR to master first.

And there are several thoughts about fixing in 6x:
* make it as a known issue (in doc) and give workaround for user (just use another sql)
* (as you said) maintain a function list in c code, and prevent them from be called in entrydb QE
* backport this PR and write a script (update the function definition) to help user to upgrade in an existed cluster

Will continue to discuss it later.

File: src/backend/catalog/cdb_schema.sql (hunk 1 above)
Comment (interma) [reply src/backend/catalog/cdb_schema.sql:51]: Indeed, they are involved in initdb process, so no previous function definition of it.

**Pull request merged**